src/controllers/CategoryFacetQueryController.ts

```diff
@@ -1,85 +1,65 @@
-    const positionInQuery = queryBuilder.categoryFacets.length;
-    this.addQueryFilter(queryBuilder, path);
-    this.addCategoryFacetRequest(queryBuilder, path, maximumNumberOfValues);
-    return positionInQuery;
+  constructor(private facet: CategoryFacet) {
+    this.resetNumberOfValuesToRequest();
+    this.resetFreezeCurrentValuesDuringQuery();
   }
 
-  public async searchFacetValues(value: string, numberOfValues: number): Promise<IGroupByValue[]> {
-    const lastQuery = { ...this.categoryFacet.queryController.getLastQuery() };
-
-    const groupByRequest: IGroupByRequest = {
-      allowedValues: [this.getAllowedValuesPattern(value)],
-      allowedValuesPatternType: AllowedValuesPatternType.Wildcards,
-      maximumNumberOfValues: numberOfValues,
-      field: this.categoryFacet.options.field as string,
-      sortCriteria: 'occurrences',
-      injectionDepth: this.categoryFacet.options.injectionDepth
-    };
-
-    lastQuery.groupBy = [groupByRequest];
-    lastQuery.categoryFacets.splice(this.categoryFacet.positionInQuery, 1);
-    const results = await this.categoryFacet.queryController.getEndpoint().search(lastQuery);
-
-    const sortByNumberOfResultsThenPathLength = (firstGroupByValue: IGroupByValue, secondGroupByValue: IGroupByValue) => {
-      if (firstGroupByValue.numberOfResults == secondGroupByValue.numberOfResults) {
-        return firstGroupByValue.value.length - secondGroupByValue.value.length;
-      }
-      return secondGroupByValue.numberOfResults - firstGroupByValue.numberOfResults;
-    };
-
-    return results.groupByResults[0].values.sort(sortByNumberOfResultsThenPathLength);
+  private resetFreezeCurrentValuesDuringQuery() {
+    this.facet.bind.onRootElement(QueryEvents.duringQuery, () => {
+      this.freezeFacetOrder = false;
+    });
   }
 
-  public addDebugGroupBy(queryBuilder: QueryBuilder, value: string) {
-    queryBuilder.groupByRequests.push({
-      field: this.categoryFacet.options.field as string,
-      allowedValues: [`.*${Utils.escapeRegexCharacter(value)}.*`],
-      allowedValuesPatternType: AllowedValuesPatternType.Regex
-    });
+  public increaseNumberOfValuesToRequest(additionalNumberOfValues: number) {
+    this.numberOfValuesToRequest += additionalNumberOfValues;
   }
 
-  private shouldAddFilterToQuery(path: string[]) {
-    return path.length != 0 && !Utils.arrayEqual(path, this.categoryFacet.options.basePath);
+  public resetNumberOfValuesToRequest() {
+    this.numberOfValuesToRequest = this.facet.options.numberOfValues;
   }
 
-  private addQueryFilter(queryBuilder: QueryBuilder, path: string[]) {
-    if (this.shouldAddFilterToQuery(path)) {
-      queryBuilder.advancedExpression.addFieldExpression(this.categoryFacet.options.field as string, '==', [
-        path.join(this.categoryFacet.options.delimitingCharacter)
-      ]);
+  /**
+   * Build the facets request for the CategoryFacet, and insert it in the query builder
```

Comment:
🤔 The class is not publically exported/documented, so why this JS Doc comment here ?